les/handler.go

```diff
@@ -1219,7 +1254,8 @@ func (pc *peerConnection) RequestHeadersByNumber(origin uint64, amount int, skip
 			return peer.GetRequestCost(GetBlockHeadersMsg, amount)
 		},
 		canSend: func(dp distPeer) bool {
-			return dp.(*peer) == pc.peer
+			p := dp.(*peer)
```

Comment:
Why did you do the same thing in two lines? Please revert if there is no reason for this.